### PR TITLE
AZP: Add PGI support from hpc-sdk module

### PR DESCRIPTION
**Pull request by avildema**
support PGI compiler from nvidia HPC-SDK toolchain

### CODE REVIEW

**Comment by yosefe** [comment]
seems not working:
image

File: buildlib/tools/common.sh [hunk 1]
```diff
@@ -54,3 +58,15 @@ prepare_build() {
...
+install_pgi() {
+	alias bc=$PGI_ROOT_DIR/../bc
+	cp $PGI_ROOT_DIR/../bc /usr/bin
+	NVHPC_ACCEPT_EULA=accept NVHPC_INSTALL_DIR=$PGI_INSTALL_DIR NVHPC_SILENT=true $PGI_ROOT_DIR/install
+	sed -i "s#<install_dir>#$PGI_INSTALL_DIR#" $PGI_MODULE
+}
+
+uninstall_pgi() {
+	sudo rm -rf $PGI_INSTALL_DIR
+}
```
Comment (yosefe) [inline buildlib/tools/common.sh:72]: newline

File: buildlib/tools/common.sh [hunk 2]
```diff
@@ -54,3 +58,18 @@ prepare_build() {
...
 }
+
+
+install_pgi() {
+	export MODULEPATH=$MODULEPATH:$WORKSPACE
+	module avail
+	mkdir -p $HOME/bin
+	# HACK: bc tool need for build hpc_sdk , we will copy it from NFS share
+	cp $PGI_ROOT_DIR/../bc $HOME/bin
+	NVHPC_ACCEPT_EULA=accept NVHPC_INSTALL_DIR=$PGI_INSTALL_DIR NVHPC_SILENT=true $PGI_ROOT_DIR/install
```
Comment (yosefe) [inline buildlib/tools/common.sh:69]: maybe just add it to PATH?

File: buildlib/tools/nvhpc-pgi-compiler [hunk 3]
```diff
@@ -0,0 +1,52 @@
+#%Module1.0
```
Comment (yosefe) [inline buildlib/tools/nvhpc-pgi-compiler:1]: 1. why we put modulefile in ucx and not in rdmz env?
2. move it to buildlib/modulefiles

File: buildlib/tools/common.sh [hunk 4]
```diff
@@ -54,3 +58,16 @@ prepare_build() {
...
+	export MODULEPATH=$MODULEPATH:$PGI_INSTALL_DIR/modulefiles
+}
+
+uninstall_pgi() {
+	rm -rf $PGI_INSTALL_DIR
+}
```
Comment (yosefe) [inline buildlib/tools/common.sh:73]: why needed? docker..

File: buildlib/tools/common.sh (hunk 2 above)
Comment (yosefe) [reply buildlib/tools/common.sh:67]: can we do smth like
export PATH=$PATH:/hpc/local/bin

File: buildlib/tools/common.sh (hunk 4 above)
Comment (avildema) [reply buildlib/tools/common.sh:73]: this dir have 6G size, and i think better remove after use
for avoid no space

Comment (yosefe) [reply buildlib/tools/common.sh:73]: isn't it gone after docker exits? or it's still cached somewhere?

**Comment by yosefe** [comment]
Seems modules fail to load:
image

File: buildlib/tools/builds.sh [hunk 5]
```diff
@@ -123,23 +123,22 @@ build_icc() {
...
+		add_network_host
 		echo "==== Build with PGI compiler ===="
 		# PGI failed to build valgrind headers, disable it for now
 		# TODO: Using non-default PGI compiler - pgcc18 which is going to be default
 		#       in next versions.
 		#       Switch to default CC compiler after pgcc18 is default for pgi module
-		${WORKSPACE}/contrib/configure-devel --prefix=$ucx_inst CC=pgcc18 --without-valgrind
+		${WORKSPACE}/contrib/configure-devel --prefix=$ucx_inst --without-valgrind
 		$MAKEP
+		make_clean distclean
```
Comment (yosefe) [inline buildlib/tools/builds.sh:138]: do_task already should remove all files, why need this too?

Comment (avildema) [reply buildlib/tools/builds.sh:138]: without it next build fail

File: buildlib/tools/common.sh [hunk 6]
```diff
@@ -25,8 +27,8 @@ parallel_jobs=4
 num_pinned_threads=$(nproc)
 [ $parallel_jobs -gt $num_pinned_threads ] && parallel_jobs=$num_pinned_threads
 
-MAKE="make"
-MAKEP="make -j${parallel_jobs}"
+MAKE="make V=1"
+MAKEP="make V=1 -j${parallel_jobs}"
```
Comment (yosefe) [inline buildlib/tools/common.sh:31]: i think we want to keep these anyway

File: buildlib/tools/common.sh [hunk 7]
```diff
@@ -9,7 +9,9 @@ GDRCOPY_MODULE="dev/gdrcopy2.1_cuda11.1.1"
 JDK_MODULE="dev/jdk"
 MVN_MODULE="dev/mvn"
 XPMEM_MODULE="dev/xpmem-90a95a4"
-PGI_MODULE="pgi/19.7"
+
```
Comment (yosefe) [inline buildlib/tools/common.sh:12]: extra space

**Review by yosefe — APPROVED** [review]

**Pull request merged**